code/modules/mob/living/basic/pets/parrot/parrot_ai/parroting_action.dm

```diff
@@ -23,28 +24,39 @@
 /datum/ai_behavior/perform_speech/parrot
 	action_cooldown = 7.5 SECONDS // gets really annoying (moreso than usual) really fast otherwise
 
-/datum/ai_behavior/perform_speech/parrot/perform(seconds_per_tick, datum/ai_controller/controller, speech, speech_sound)
+/datum/ai_behavior/perform_speech/parrot/perform(seconds_per_tick, datum/ai_controller/controller, list/speech, speech_sound)
 	var/mob/living/basic/parrot/speaking_pawn = controller.pawn
 	var/list/available_channels = speaking_pawn.get_available_channels()
-	var/modified_speech = speech
+	var/modified_speech = speech["line"]
 	var/use_radio = prob(50) // we might not even use the radio if we even have a channel
 
-#define HAS_CHANNEL_PREFIX (speech[1] in GLOB.department_radio_prefixes) && (copytext_char(speech, 2, 3) in GLOB.department_radio_keys) // determine if we need to crop the channel prefix
+	var/has_channel_prefix = (modified_speech[1] in GLOB.department_radio_prefixes) && (copytext_char(modified_speech, 2, 3) in GLOB.department_radio_keys) // determine if we need to crop the channel prefix
 
 	if(!length(available_channels)) // might not even use the radio at all
-		if(HAS_CHANNEL_PREFIX)
-			modified_speech = copytext_char(speech, 3)
+		if(has_channel_prefix)
+			modified_speech = copytext_char(modified_speech, 3)
 
 	else
-		if(HAS_CHANNEL_PREFIX)
-			modified_speech = "[use_radio ? pick(available_channels) : ""][copytext_char(speech, 3)]"
+		if(has_channel_prefix)
+			modified_speech = "[use_radio ? pick(available_channels) : ""][copytext_char(modified_speech, 3)]"
 		else
-			modified_speech = "[use_radio ? pick(available_channels) : ""][speech]"
-
+			modified_speech = "[use_radio ? pick(available_channels) : ""][modified_speech]"
 
+	if(SStts.tts_enabled && SStts.available_speakers.Find(speech["voice"]))
+		speaking_pawn.voice = speech["voice"]
+	if(speech["pitch"] && SStts.pitch_enabled)
+		speaking_pawn.pitch = min(speech["pitch"] + rand(12, 24), MAXIMUM_PARROT_PITCH)
```

Comment:
Split out all of the TTS stuff here into a new proc that we only invoke if `if(SStts.tts_enabled)` is `TRUE`, I don't want something that's optional for a server to have be oddly in the middle what's intended to be a relatively straight-forward AI behavior.